lib/solr/core_configuration/core.rb

```diff
@@ -0,0 +1,34 @@
+module Solr
+  module CoreConfiguration
+    class Core
```

Comment:
maybe for simpliciity we'll use `core` for user-facing APIs (method args etc) and for internal configuration use `CoreConfig` to distinguish it from the stringy core that's actually is a core_name?

Reply:
I already suggested renaming `core_name` to `core`: https://github.com/machinio/solrb/pull/12#issuecomment-414370224
Will do that.